File: 2022/01/go/1b.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)

func findTop3Calories() []int {
	readFile, err := os.Open("../input.txt")
	singleElfTotal := 0
	totalSlice := []int{}

	if err != nil {
		fmt.Println(err)
	}

	fileScanner := bufio.NewScanner(readFile)

	fileScanner.Split(bufio.ScanLines)

	for fileScanner.Scan() {
		currentLine := fileScanner.Text()
		if currentLine == "" {
			totalSlice = append(totalSlice, singleElfTotal)
			singleElfTotal = 0
		} else {
			currentLineValue, _ := strconv.Atoi(currentLine)
			singleElfTotal += currentLineValue
		}
	}

	readFile.Close()
	sort.Slice(totalSlice, func(i int, j int) bool { return totalSlice[i] > totalSlice[j] })
	return totalSlice[:3]
}

func one_b() {
	top3CaloriesSum := 0
	totalArray := findTop3Calories()

	for _, value := range totalArray {
		top3CaloriesSum += value
	}

	fmt.Println(top3CaloriesSum) // 200158
}

File: 2022/01/python/1b.py

def find_top3_calories():
    single_elf_total = 0
    total_array = []
    with open("../input.txt") as f:
        for line in f:
            if line == "\n":
                total_array.append(single_elf_total)
                single_elf_total = 0
            else:
                single_elf_total += int(line)

    total_array.sort(reverse=True)

    return total_array[:3]

top_three_calories = find_top3_calories()
print(sum(top_three_calories)) # 200158
